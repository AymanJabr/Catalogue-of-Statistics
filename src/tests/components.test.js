import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import Header from '../components/header';
import SearchBar from '../components/searchBar';
import CountryCard from '../components/countryCard';
import WorldwideCard from '../components/worldwideCard';
import Pagination from '../components/pagination';

configure({ adapter: new Adapter() });

describe('Header', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Pagination', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Pagination
      totalPages={5}
      pageNumber={2}
      goToPreviousPage={() => true}
      goToNextPage={() => true}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('SearchBar', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SearchBar searchCountry={() => true} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('CountryCard', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<CountryCard />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('WorldwideCard', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<WorldwideCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
