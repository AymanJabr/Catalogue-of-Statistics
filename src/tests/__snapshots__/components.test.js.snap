// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryCard renders correctly 1`] = `
<div
  className="countryCard "
>
  <Link
    to="/country/null"
  >
    <img
      alt="null"
      className="countryCardImage"
      src="./countries_flags/null.png"
    />
    <h1
      className="countryCardName"
    />
    <p
      className="countryCardCasesNumber"
    >
      active cases:
       0
    </p>
    See all country statistics
  </Link>
</div>
`;

exports[`Header renders correctly 1`] = `
<div
  className="Header"
>
  <div
    className="Navigation "
  >
    <Link
      to="/"
    >
      Go back to all countries  
    </Link>
  </div>
</div>
`;

exports[`SearchBar renders correctly 1`] = `
<div
  className="SearchBar"
>
  <img
    alt="Search"
    className="SearchIcon"
    src="data:image/svg+xml;utf8;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iaXNvLTg4NTktMSI/Pgo8IS0tIEdlbmVyYXRvcjogQWRvYmUgSWxsdXN0cmF0b3IgMTkuMC4wLCBTVkcgRXhwb3J0IFBsdWctSW4gLiBTVkcgVmVyc2lvbjogNi4wMCBCdWlsZCAwKSAgLS0+CjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgdmVyc2lvbj0iMS4xIiBpZD0iQ2FwYV8xIiB4PSIwcHgiIHk9IjBweCIgdmlld0JveD0iMCAwIDU2Ljk2NiA1Ni45NjYiIHN0eWxlPSJlbmFibGUtYmFja2dyb3VuZDpuZXcgMCAwIDU2Ljk2NiA1Ni45NjY7IiB4bWw6c3BhY2U9InByZXNlcnZlIiB3aWR0aD0iMTZweCIgaGVpZ2h0PSIxNnB4Ij4KPHBhdGggZD0iTTU1LjE0Niw1MS44ODdMNDEuNTg4LDM3Ljc4NmMzLjQ4Ni00LjE0NCw1LjM5Ni05LjM1OCw1LjM5Ni0xNC43ODZjMC0xMi42ODItMTAuMzE4LTIzLTIzLTIzcy0yMywxMC4zMTgtMjMsMjMgIHMxMC4zMTgsMjMsMjMsMjNjNC43NjEsMCw5LjI5OC0xLjQzNiwxMy4xNzctNC4xNjJsMTMuNjYxLDE0LjIwOGMwLjU3MSwwLjU5MywxLjMzOSwwLjkyLDIuMTYyLDAuOTIgIGMwLjc3OSwwLDEuNTE4LTAuMjk3LDIuMDc5LTAuODM3QzU2LjI1NSw1NC45ODIsNTYuMjkzLDUzLjA4LDU1LjE0Niw1MS44ODd6IE0yMy45ODQsNmM5LjM3NCwwLDE3LDcuNjI2LDE3LDE3cy03LjYyNiwxNy0xNywxNyAgcy0xNy03LjYyNi0xNy0xN1MxNC42MSw2LDIzLjk4NCw2eiIgZmlsbD0iIzAwMDAwMCIvPgo8Zz4KPC9nPgo8Zz4KPC9nPgo8Zz4KPC9nPgo8Zz4KPC9nPgo8Zz4KPC9nPgo8Zz4KPC9nPgo8Zz4KPC9nPgo8Zz4KPC9nPgo8Zz4KPC9nPgo8Zz4KPC9nPgo8Zz4KPC9nPgo8Zz4KPC9nPgo8Zz4KPC9nPgo8Zz4KPC9nPgo8Zz4KPC9nPgo8L3N2Zz4K"
  />
  <input
    className="Search"
    id="header-search"
    name="header"
    onChange={[Function]}
    placeholder="Search country"
    type="text"
  />
</div>
`;
